### PR TITLE
Rename EZB resolver driver to JOP, and improve code

### PR DESCRIPTION
**Pull request by RLangeUni**
* rename ezb to jop for clarity reasons
* make xpath title selector a static variable that can be overwritten for better inheritance
* add result state 10 for Unknown Print / Unknown Electronics
* check date format in downgradeOpenUrl

TODO
- [x] Update changelog when merging (note renamed driver)

### CODE REVIEW

File: module/VuFind/src/VuFind/Resolver/Driver/PluginManager.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Resolver/Driver/PluginManager.php:53]: We should set up some aliases from ezb and the old class name to point to the new jop name, so people with outdated configurations won't break their resolving.

**Review by demiankatz — CHANGES_REQUESTED** [review]
One quick suggestion regarding backward compatibility -- once that's fixed, I'll take a closer look at the rest of the code changes (I'm a little short on time today, but I'll have more availability next week). Thanks for this!

**Comment by RLangeUni** [comment]
I have added an extra alias for the ol ezb resolver name.

File: module/VuFind/src/VuFind/Resolver/Driver/Jop.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: Is there a reason we pass back the original settings if we were unable to find anything to downgrade? This seems like a behavior change from before, and I'm not sure that it will end up with good results. If there's a reason for it, maybe a comment would help explain why.

File: module/VuFind/src/VuFind/Resolver/Driver/Jop.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: Is there a strong reason to make this static, rather than a regular property?

**Review by demiankatz — COMMENTED** [review]
Thanks, @RLangeUni! I gave this a closer review now and had a couple of small questions.

I'll also ask around and see if there are any other EZB/JOP users who might have input on this.

Since this is a backward-breaking change, I've marked this for the 9.0 milestone and will plan to merge it after 8.1 is released (or into the dev-9.0 branch, if it's ready before that happens).

I've also added a TODO checkbox to the description as a reminder to update the changelog to note the driver name change when we merge this.

**Comment by demiankatz** [comment]
Quick update: I spoke with Markus Fischer, who has worked on the EZB driver in the past, and he approves the changes... so I think we can merge this once the small questions I asked above have been addressed.

File: module/VuFind/src/VuFind/Resolver/Driver/Jop.php
Comment (RLangeUni) [reply module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: Yes, you were right. I fixed that. Thanks.

File: module/VuFind/src/VuFind/Resolver/Driver/Jop.php
Comment (RLangeUni) [reply module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: In finc we are using a trait and a child class of EZB / JOP with a constructor, in which we do not call the parent constructor. So with a static property or const we would minimize changes.

Comment (demiankatz) [reply module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: @RLangeUni, as long as you initialize the property directly in the property declaration rather than in the contructor, the value should be correctly populated whether or not the parent constructor gets explicitly called. The property doesn't need to be static for this to work. (Unless I'm misunderstanding something).

If you want to be able to explicitly refer to Jop::$xpathTitleSelector from a trait, as opposed to having the trait simply assume that $this->xpathTitleSelector is set, maybe it would be preferable to use a constant rather than a static property.

Anyway, if you still prefer to keep this as-is, I don't mind -- but I just want to be absolutely sure it shouldn't be simplified before I merge it. :-)

File: module/VuFind/src/VuFind/Resolver/Driver/Jop.php
Comment (demiankatz) [reply module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: Thanks!

File: module/VuFind/src/VuFind/Resolver/Driver/Jop.php
Comment (robertlange81) [reply module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: Yes, You were right. I might have thought to cumbersome and can no longer understand my thoughts from back then, so I made xpathTitleSelector a regular property like the others.

Comment (demiankatz) [reply module/VuFind/src/VuFind/Resolver/Driver/Jop.php]: Great, thanks! I believe this is ready to merge now.

**Review by demiankatz — APPROVED** [review]

**Pull request merged**
